tests/test_weighted_random_sampler.py

```diff
@@ -155,3 +152,25 @@ def test_segmented_random_sampler_simple_distribution(seed, num_samples, precisi
     _, counts = np.unique(test_selection, return_counts=True)
     percentages = counts / num_samples
     np.testing.assert_allclose(expected_percentages, percentages, atol=0.05)
+
+
+@pytest.mark.parametrize("seed", [2024])
+@pytest.mark.parametrize("num_samples", [1000])
+@pytest.mark.parametrize("precision", [np.float32, np.float64])
+def test_segmented_random_sampler_zero_probability(seed, num_samples, precision):
+    """Make sure that if we zero out a probability we never sample that value"""
+
+    # Setup weights such that expected percentages are obvious
+    weights = [[0.0, 2.5] for _ in range(num_samples)]
```

Comment:
nit: seems like elsewhere we use "weights" to mean "log probabilities"?
```suggestion
    probabilities = [[0.0, 2.5] for _ in range(num_samples)]
```

Reply:
I think by https://github.com/proteneer/timemachine/pull/1240#issuecomment-1909145964 we are using the convention of:

* weights: unnormalized probabilities (positive sum but doesn't have to be 1), non-log space. The Exchange movers are imprecise in calling log weights 'weights', can adjust
* log-weights: log space weights
* probabilities: normalized weights that sum to 1.0

Reply:
Ah, got it, thanks!

Reply:
Looking the weights in tibd/bd need to be cleaned up in a more specific way then I had thought. Going to put that to another PR to avoid mixing up orthogonal changes.

Reply:
Nevermind, misunderstood the code, renamed the variables to clarify that they are log weights in https://github.com/proteneer/timemachine/pull/1240/commits/1692502dbbcf6544b8752ec63a38e84bdeff1a9a